locator/token_metadata.cc

```diff
@@ -1290,6 +1290,19 @@ generic_token_metadata<NodeId>::get_all_endpoints() const {
     return _impl->get_all_endpoints();
 }
 
+template <typename NodeId>
+template <typename T>
+requires std::is_same_v<T, locator::host_id>
+std::unordered_set<gms::inet_address> generic_token_metadata<NodeId>::get_all_ips() const {
+    const auto& host_ids = _impl->get_all_endpoints();
+    std::unordered_set<gms::inet_address> result;
+    result.reserve(host_ids.size());
+    for (const auto& id: host_ids) {
+        result.insert(_impl->get_endpoint_for_host_id(id));
```

Comment:
In my mind, the idea of making `token_metadata` to be host-ID based was that token_metadata and its internals were on longer aware of IP addresses (they wouldn't even "understand" the concept of IP addresses); there would be a translation layer **around** (outside) `token_metadata` when interfacing with code that requires IP addresses.

But here it looks like we're not there yet... the IPs are still living "inside" token_metadata.
That is, token_metadata stores a mapping of IDs to IPs inside itself and sometimes uses it inside its implementations. (`get_endpoint_for_host_id`)

I think we'll need to get rid of this mapping eventually. (Probably in a follow-up.)